pysteps/nowcasts/steps.py

```diff
@@ -33,6 +33,873 @@
     DASK_IMPORTED = False
 
 
+class StepsNowcaster:
+    def __init__(self, precip, velocity, timesteps, **kwargs):
+        # Store inputs and optional parameters
+        self.precip = precip
```

Comment:
I am wondering if it would be possible to pass only the "scaffolding" here (what are now the kwargs) and then pass `precip`, `velocity` and `timesteps`, when you actually run the nowcast. This way you could create this class once and reuse it for every new iteration.

Reply:
So I see how we could do this but the old functionality of forecast() can not be changed. Only when people want to use the new framework more modularly would this be possible. In this case forecast would look something like this:
- init StepsNowcasterConfig
- init nowcaster (with precip, velocity, timesteps, and Config)
- nowcaster.compute_forecast()
This would enable people who run the forecast for multiple run times to define the StepsNowcasterConfig once and then loop over all times. This is the only solution I see right now but this would not be possible in the current forecast() method.

Reply:
Yes the old `forecast()` method could indeed not be updated to this new way of running the code, but maybe we can deprecate the old `forecast()` method over time? Actually this could easily be done when the xarray v2 would be released.

As for how this would work, I was thinking something like this:
```
nowcaster = StepsNowcast(timesteps, StepsNowcastConfig(...))
nowcaster.compute_forecast(precip, velocity)
```
The `__init__` could actually already take care of some things that don't need `precip` and `velocity` like retrieving the oflow method to be used.

The only thing you need to be very careful with in this case is that you reset the state properly every time `compute_forecast` is called.

Reply:
I am going step by step. This Config file is now added; please let me know if this is how you want me to continue or if I need to change the code. Currently, the config file is a class variable. All features from these variables are always accessed as follows: self.config.XXXX. That way, it is always clear what parameters are configuration parameters and which ones are parameters or states for the nowcaster. The next step is to add these two...